### PR TITLE
Release 3.4.1

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by kenibrewer — COMMENTED** [review]
This looks really good to me. I have a few optional requests for some comments and non-actionable discussion items but otherwise this looks ready to release once we have passing tests.

File: main.nf
[the remark this answers is not in the document]
Comment (maxulysse) [reply main.nf:1]: yes, that was intentional, actually one can launch a nextflow pipeline without the `nextflow run` command.

File: subworkflows/local/annotation_cache_initialisation/main.nf [hunk 1]
```diff
@@ -1,14 +1,14 @@
 //
-// INITIALIZE ANNOTATION CACHE
+// ANNOTATION CACHE INITIALISATION
 //
 
-// Initialize channels based on params or indices that were just built
+// Initialise channels based on params or indices that were just built
```
[the remark this answers is not in the document]
Comment (maxulysse) [reply subworkflows/local/annotation_cache_initialisation/main.nf:5]: It was more to be in check with the other `initialise` subworkflows, I honestly don't care much about en-GB vs en-US spelling.

File: subworkflows/local/utils_nfcore_sarek_pipeline/main.nf [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+            case 'prepare_recalibration':   log.warn("Using file ${outdir}/csv/markduplicates_no_table.csv");
+                                            input = outdir + "/csv/markduplicates_no_table.csv"
+                                            break
+            case 'recalibrate':             log.warn("Using file ${outdir}/csv/markduplicates.csv");
+                                            input = outdir + "/csv/markduplicates.csv"
+                                            break
+            case 'variant_calling':         log.warn("Using file ${outdir}/csv/recalibrated.csv");
+                                            input = outdir + "/csv/recalibrated.csv"
+                                            break
+            // case 'controlfreec':         csv_file = file("${outdir}/variant_calling/csv/control-freec_mpileup.csv", checkIfExists: true); break
```
[the remark this answers is not in the document]
Comment (maxulysse) [reply subworkflows/local/utils_nfcore_sarek_pipeline/main.nf:350]: It's a past feature that we want to restore at some point, good spot

File: tests/config/pytesttags.yml
[the remark this answers is not in the document]
Comment (maxulysse) [reply tests/config/pytesttags.yml:1]: that's the whole idea there, converting all pytest tests to nf-test tests

**Comment by maxulysse** [comment]
All missing md5sum are from pytest tests that we decided to slowly depreciate and as md5sum are way more complicated to update, if there's any issue, we don't update anymore until we move to nf-test

**Review by kenibrewer — APPROVED** [review]
@maxulysse Thanks for the responses. LGTM

File: main.nf [hunk 3]
```diff
@@ -20,103 +20,357 @@
...
 
-// WORKFLOW: Execute a single named workflow for the pipeline
-// See: https://github.com/nf-core/rnaseq/issues/619
-workflow {
-    NFCORE_SAREK ()
+//
+// Get attribute from genome config file e.g. fasta
+//
+
+def getGenomeAttribute(attribute) {
```
Comment (svarona) [inline main.nf:367]: It doesn't matter if `getGenomeAttribute `is in the `main.nf` or in `subworkflows/nf-core/utils_nextflow_pipeline/main.nf` ?

Comment (maxulysse) [reply main.nf:367]: @ramprasadn and I noticed some issues with the `getGenomeAttribute` from `subworkflows/nf-core/utils_nextflow_pipeline/main.nf` , so we're trying out a local one in the `main.nf`

**Pull request merged**